src/internal.h

```diff
@@ -138,6 +138,26 @@
 #endif
 
 /**
+* AUGEAS_LIKELY:
+*
+* Macro to flag a code branch as a likely branch
+*
+* AUGEAS_UNLIKELY:
+*
+* Macro to flag a code branch as an unlikely branch
+*/
+#ifndef __has_builtin
+#   define __has_builtin(x) (0)
+#endif
+#if defined(__builtin_expect) || __has_builtin(__builtin_expect)
+#   define AUGEAS_LIKELY(x)        (__builtin_expect(!!(x), 1))
```

Comment:
Could we just call them `LIKELY` and `UNLIKELY` or is that going to clash with some other include ?

Reply:
These macros are always prefixed by the project name since other projects may also define their own set of builtin_expect macros.